backend/inspirehep/alembic/5a0e2405b624_add_citation_type_to_citation_table.py

```diff
@@ -0,0 +1,30 @@
+#
+# This file is part of Invenio.
+# Copyright (C) 2016-2018 CERN.
+#
+# Invenio is free software; you can redistribute it and/or modify it
+# under the terms of the MIT License; see LICENSE file for more details.
+
+"""Add is_self_citation column to citation_table"""
+
+import sqlalchemy as sa
+from alembic import op
+
+# revision identifiers, used by Alembic.
+revision = "5a0e2405b624"
+down_revision = "595c36d68964"
+branch_labels = ()
+depends_on = None
+
+
+def upgrade():
+    """Upgrade database."""
+    op.add_column(
+        "records_citations",
+        sa.Column("is_self_citation", sa.Boolean, nullable=True, default=False),
```

Comment:
When we will Create this column it will be null at start. Setting default value in there won't help. To overcome this we would havge first to create migration with null=True then migrate ALL citations with some value, then apply another migration which will disallow null values.

It can be done in one migration with something like:
```python
op.add_column(
    "records_citations",
    sa.Column("is_self_citation", sa.Boolean, nullable=True, default=False)
)
op.execute("UPDATE records_citations SET is_self_citation = false")
op.alter_column('records_citations', 'is_self_citation', nullable=False)
```

But this migration would take some time and during this process continuous_migration, record updates, indexing, etc would be broken and won't work at all :/

Reply:
in the update functions you assume that's it's either set to `true` or `false` for existing record. So I believe nothing will be updated for those records.

Reply:
Until first migration will be done they will be just ignored. That's true. But it won't break anything and will still work properly with FEATURE_FLAG set to False